eyeshade/workers/snapshots.js

```diff
@@ -0,0 +1,94 @@
+const Cursor = require('pg-cursor')
+const uuidV5 = require('uuid/v5')
+const { BigNumber } = require('bat-utils/lib/extras-utils')
+
+exports.initialize = async (debug, runtime) => {
+  await runtime.queue.create('update-snapshot-accounts')
+}
+
+const getAllAccountBalances = `
+select * from account_balances
```

Comment:
I think ideally we would be able to generate account balances for any period in time (including the past).  This query only supports generating snapshots at the given moment.  So I think if we bypass account_balances and go straight to account_transactions, we can do this.

This is a query I use to generate ads account balances precisely on the first one the month - https://github.com/brave-intl/gordon/blob/b8e52fb77244a1de0816bd073bb5646236f0b490/bin/create-payout-report-ads-v2.py#L16.  I think some version of that query can work here.  Then we can let publishers (or whatever client ends up creating these snapshots) determine at what time they want the balances to be frozen at.  But I wouldn't focus too much on the API for now

Reply:
yeah, started on this aspect on another revision of this. was going to run a test with prod data locally to see how slow queries would be